plugins/source.go

```diff
@@ -119,18 +119,17 @@ func (p *SourcePlugin) TablesForSpec(spec specs.Source) (schema.Tables, error) {
 	if err := spec.Validate(); err != nil {
 		return nil, fmt.Errorf("invalid spec: %w", err)
 	}
-	tables, err := p.listAndValidateTables(spec.Tables, spec.SkipTables)
+	selectedTables, err := p.listAndValidateTables(spec.Tables, spec.SkipTables)
 	if err != nil {
 		return nil, err
 	}
-	// listAndValidateTables returns a flattened list - we only want to return
-	// the top-level tables from this function.
+	// listAndValidateTables returns a set of all tables - we only want to return
+	// the top-level tables from the set in this function.
```

Comment:
Technically I think this behavior should change, but I'm not changing it in this PR.